css-easing-1/Overview.bs

```diff
@@ -125,16 +125,126 @@ The syntax for specifying an [=easing function=] is as follows:
+     and [=linear easing point/output=] as the y-axis.
+
+1.   Interpolate the points linearly.
+
+1.   Extend the ends of the graph to infinity using the final angle of the line at either end.
+     If the line starts/ends with two points of the same position, the line should extend along the x-axis.
+
+1.   Return the y-axis value corresponding to the x-axis value of |inputProgress|.
+
+</section>
+
+<dfn function lt="linear()">linear(<<linear-stop-list>>)</dfn>
+
+<pre class="prod">
+  <dfn>&lt;linear-stop-list></dfn> = [ <<linear-stop>> ]#
+  <dfn>&lt;linear-stop></dfn> = <<number>> && <<linear-stop-length>>?
+  <dfn>&lt;linear-stop-length></dfn> = <<percentage>>{1,2}
+</pre>
+
+<section algorithm="to create a linear easing function">
+
+To <dfn>create a linear easing function</dfn>
+from a <<linear-stop-list>> |stopList|,
+perform the following.
+It returns a [=linear easing function=].
+
+1.   Let |function| be a new [=linear easing function=].
+
+1.   Let |largestInput| be negative infinity.
+
+1.   For each |stop| in |stopList|:
+
+     1.   Let |point| be a new [=linear easing point=].
+
+     1.   [=list/Append=] |point| to |function|'s [=linear easing function/points=].
+
+     1.   Set |point|'s [=linear easing point/output=] to |stop|'s <<number>>.
+
+     1.   If |stop| has a <<linear-stop-length>>, then:
+
+          1.   Set |point|'s [=linear easing point/input=] to whichever is greater:
+               |stop|'s <<linear-stop-length>>'s first <<percentage>> as a number,
+               or |largestInput|.
+
+          1.   Set |largestInput| to |point|'s [=linear easing point/input=].
+
+          1.   If |stop|'s <<linear-stop-length>> has a second <<percentage>>, then:
+
+               1.   Let |extraPoint| be a new [=linear easing point=].
+
+               1.   [=list/Append=] |extraPoint| to |function|'s [=linear easing function/points=].
+
+               1.   Set |extraPoint|'s [=linear easing point/output=] to |stop|'s <<number>>.
+
+               1.   Set |extraPoint|'s [=linear easing point/input=] to whichever is greater:
+                    |stop|'s <<linear-stop-length>>'s second <<percentage>> as a number,
+                    or |largestInput|.
+
+               1.   Set |largestInput| to |extraPoint|'s [=linear easing point/input=].
+
+     1.   Otherwise, if |stop| is the first [=list/item=] in |stopList|,
+          then set |point|'s [=linear easing point/input=] to 0.
+
+     1.   Otherwise, if |stop| is the last [=list/item=] in |stopList|,
+          then set |point|'s [=linear easing point/input=] to 1.
+
+1.   For runs of [=list/items=] in |function|'s [=linear easing function/points=] that have a null [=linear easing point/input=],
+     assign a number to the [=linear easing point/input=] by linearly interpolating between the previous and next [=linear easing function/points=] that have a non-null [=linear easing point/input=].
+
+1.   Return |function|.
```

Comment:
Serializing back from this returned function may result in a different round trip result: `linear(0 0% 50%, 1 50% 100%)` -> `linear(0 0%, 0 50%, 1 50%, 1 100%)`.  Do we want to specify how the serialization should be carried out?

Reply:
What happens with linear-gradient? If there's prescient there, same should happen here.

Reply:
`linear-gradient(0deg, red 0% 10%, green, blue 90%);` seems to become `llinear-gradient(0deg, rgb(255, 0, 0) 0%, rgb(255, 0, 0) 10%, rgb(0, 128, 0), rgb(0, 0, 255) 90%);`.
That said, not sure if I 100% agree on keeping it the same as `linear-gradient`:
* There may be a lot more stops being supplied for the easing function in general
* `linear-gradient`'s `function` equivalent is really called once for paint vs. every frame of animation for this easing function. 

So the two options of keeping both presentable & function forms, or calculating the function form every time seem... unattractive.

Reply:
I agree it's a little weird that syntax unfolds like that, but I think it still makes sense to do the same as gradient.